build_miniforge_osx.sh

```diff
@@ -4,14 +4,31 @@ set -e
 set -x
 
 echo "Installing a fresh version of Miniforge3."
-MINIFORGE_URL="https://github.com/conda-forge/miniforge/releases/download/4.8.3-1"
-MINIFORGE_FILE="Miniforge3-4.8.3-1-MacOSX-x86_64.sh"
-curl -L -O "${MINIFORGE_URL}/${MINIFORGE_FILE}"
-bash "${MINIFORGE_FILE}" -b
+# Keep variable names in sync with 
+# https://github.com/conda-forge/docker-images/blob/main/scripts/run_commands
+miniforge_arch="$(uname -m)"
+miniforge_version="4.10.3-10"
+condapkg="https://github.com/conda-forge/miniforge/releases/download/${miniforge_version}/Mambaforge-${miniforge_version}-MacOSX-${miniforge_arch}.sh"
+if [ "$(uname -m)" = "x86_64" ]; then
+   # hmaarrfk: 2021/12/01 Not too sure how to check sha256sums on OSX
+   # shellcheck disable=SC2034
+   conda_chksum="7c44259a0982cd3ef212649678af5f0dd4e0bb7306e8fffc93601dd1d739ec0b"
+elif [ "$(uname -m)" = "arm64" ]; then
```

Comment:
@erykoff is this correct for macOS ARM? Or should we have something else here?

Reply:
Running `uname -m` on my machine does indeed say `arm64`.

Reply:
Thanks for checking! 😄